hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -98,4 +105,35 @@ public void testMemberConnectionOrder() {
         Collection<Client> connectedClients2 = server2.getClientService().getConnectedClients();
         assertEquals(connectedClients2.size(), 0);
     }
+
+    @Test
+    public void testConnectionRemovedOnce() {
+        HazelcastInstance server = Hazelcast.newHazelcastInstance();
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        HazelcastClientInstanceImpl clientImpl = ClientTestUtil.getHazelcastClientInstanceImpl(client);
+        ClientConnectionManager connectionManager = clientImpl.getConnectionManager();
+
+
+        final AtomicInteger count = new AtomicInteger();
+
+        connectionManager.addConnectionListener(new ConnectionListener() {
+            @Override
+            public void connectionAdded(Connection connection) {
+
+            }
+
+            @Override
+            public void connectionRemoved(Connection connection) {
+                count.incrementAndGet();
+            }
+        });
+
+        final Address serverAddress = new Address(server.getCluster().getLocalMember().getSocketAddress());
+        final Connection connectionToServer = connectionManager.getConnection(serverAddress);
```

Comment:
I am adding the CountingConnectionListener by the way. (as CountingConnectionRemoveListener. because the reader can misunderstand since code is elsewhere, I need to define what it clearly doing in the name)